R/utils.R

```diff
@@ -0,0 +1,16 @@
+globalVariables(
+  c(".", "County", "Name", "Prediction", "Interval", "State", "StateFIPS", 
+    "V1", "V2", "abbreviation", "alpha", "cases", "cases_deaths", "county", 
+    "countyFIPS", "cum", "deaths", "designation", "endpoint_type", 
+    "exists_interval_score_0", "fips", "forecast_date", "full_location_name", 
+    "geo_type", "head", "horizon", "inc", "inc_cases", "inc_deaths", 
+    "interval_score_0", "location", "location_name", "lower", "model", 
+    "model.x", "model_abbr", "n_interval_scores", "notes", "overprediction_0", 
+    "point", "quantile", "sharpness_0", "state", "state_abbreviation", 
+    "tail", "target", "target_end_date", "target_variable", 
+    "team_model_designation", "temporal_resolution", "timezero", "true_value", 
+    "truth_forecast", "truth_model", "type", "underprediction_0", 
+    "unit", "upper", "value", "value.x", "value.y", 
+    "County", "Name", "Prediction", "Interval"
+  )
+)
```

Comment:
It seems like these are column names in various data frames that are referenced within tidyverse operations like `dplyr::mutate`, and not actually global variables defined by the package.  For example, the variables like `overprediction_0`, `sharpness_0`, and `underprediction_0` are referenced at https://github.com/reichlab/covidHubUtils/blob/b4e5fa21d878c8c95d21745191346a499e6e8495/R/score_forecasts.R#L157.  It seems like declaring them as `globalVariables` is probably not the right thing to do here.  Should we delete this?  If you were prompted to create this by something like `devtools::check()` or similar, does that suggest we should make some other change at the point these variables are referenced to satisfy the check?

Reply:
I found this thread which suggests all those mutate statements etc should use `.data$underprediction_0` and similar, after importing `.data` from `rlang`: https://community.rstudio.com/t/when-programming-with-dplyr-what-is-the-correct-way-to-avoid-undefined-global-variables/55946

Maybe we could delete this `globalVariables` call for now and create a separate issue to clean those up?

Reply:
hm I always defined them as globalVariables to get rid of the note (and also have seen that elsewhere in other packages), but I admit I don't _really_ _really_ understand what is going on underneath the hood. So happy to do it either way. Would you like me to make a change? Or do you just want to change it yourself?

Reply:
Maybe @seabbs has smart thoughts?

Reply:
I think this is not a big deal, so I'd be fine with merging this in for now and filing a separate issue for later to go back and revise the calls to mutate

Reply:
:+1: sounds good to me